core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -401,17 +403,24 @@ private[spark] class TaskSchedulerImpl(
       val host = shuffledOffers(i).host
       val taskSetRpID = taskSet.taskSet.resourceProfileId
 
+      var continueScheduling = true
       // check whether the task can be scheduled to the executor base on resource profile.
-      if (sc.resourceProfileManager
-        .canBeScheduled(taskSetRpID, shuffledOffers(i).resourceProfileId)) {
+      while (sc.resourceProfileManager
```

Comment:
I'm not sure I like this coding approach of just short circuiting here and bypassing everything of the caller resourceOffers function.

I think this approach is going to superceded locality for instance. In the very least we need to define that behavior.  Was your intention to override locality?
there are other obvious things like the resourceOffers comment is now wrong because it mentions round robin and this is overriding that.

I'm also curious if this has other side affects like potentially FAIR Scheduler. its something I haven't looked at in a long time so need to spend more time on it.

Reply:
Agree with @tgravescs, this PR does violate scheduling policies and is assuming FIFO

Reply:
Scheduling policy such as FIFO or Fair is using single task set as schedule unit. My change is inside `resourceOfferSingleTaskSet`, which not change how task set is scheduled.

Reply:
There is a behavior difference between how many asks get used up for the first taskset before it moves to the next.